docs/en/development/developing-component.md

```diff
@@ -0,0 +1,139 @@
+## 1、Preparation Work
+
+1. Git clone the repository to your preferred directory
+2. Use Docker to launch the environment you need. For example, if you want to develop a distributed lock API with ZooKeeper, you need to start a ZooKeeper container locally with Docker for local tests.
+   If you do not have Docker locally, you can install a Docker Desktop by following [Docker Desktop tutorial](https://www.runoob.com/docker/windows-docker-install.html). Mac and Windows are both supported, easy to use.
+   
+## 2、Development Components and Unit Tests
+### 2.1.Create a new folder under Components/API directory to develop your components
+
+The folder name can use the component name, referring to the redis component below
+
+![img.png](../../img/development/component/img.png)
+
+Potential tools you may use in the process of development (for the purpose of reference only, hope to simplify the development) :
+
+- Once staring a new goroutine, panic triggered in it may lead to a panic breakdown to the entire server. Therefore, it is common to start a goroutine with recover() inside deferred functions. You can also use the encapsulated utility classes, like utils.GoWithRecover in mosn.io/pkg/utils/goroutine.go
+
+- log.DefaultLogger is a commonly used logging tool, and it is located in mosn.io/pkg/log/errorlog.go
+
+- Others: There are utility classes under pkg/common and mosn.io/pkg directory
+
+### 2.2. Copy and Paste Other Components
+
+You can simply copy and paste other components for modification or development. For example, if you want to implement the distributed lock API using ZooKeeper, you can copy and paste the Redis component
+
+### 2.3. Write Unit Tests!
+#### 2.3.1. Unit testing considerations
+Unit tests will be run in various environments, including the docker provided by github action and other developers' computers. Thus, following problems need to be considered to run unit tests normally:
+- Other people may not have ZooKeeper installed in their environments. So when we write a unit test, either mock out the network call code (for example, mock out the part of the ZooKeeper code in unit tests) or create a simple version of ZooKeeper in the unit test (for example, in the Redis Unit test, A mini-redis will be created) to ensure others can pass the test.
+- Since every time someone commits code, it automatically runs unit tests, and they are merged only when they are all passed. Therefore, try to avoid sleeping too long in the unit test (sleeping too long can decrease the speed for unit tests)
+#### 2.3.2. How to mock out dependencies in the environment once running unit tests? (Such as Mock ZooKeeper or Mock Redis)
+
+It is usually to encapsulate all network call code into a single interface, and then mock out that interface in ut. Taking the unit tests in Apollo configuration center as an example, and refer to components/configstores/apollo/configstore.go. and
```

Comment:
It is usual  to ... not a usually here which is a adverb.

"Take the unit tests in Apollo configuration center as an example, referring to components/configstores/apollo/configstore.go. "
is better than
"Taking the unit tests in Apollo configuration center as an example, and refer to components/configstores/apollo/configstore.go. "

Reply:
fixed